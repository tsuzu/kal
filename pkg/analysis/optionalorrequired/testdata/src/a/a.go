package a

type OptionalOrRequiredTestStruct struct {
	NoMarkers string // want "field NoMarkers must be marked as optional or required"

	// noOptionalOrRequiredMarker is a field with no optional or required marker.
	// +enum
	// +kubebuilder:validation:Enum=Foo;Bar
	NoOptionalOrRequiredMarker string // want "field NoOptionalOrRequiredMarker must be marked as optional or required"

	// +optional
	// +required
	MarkedOpitonalAndRequired string // want "field MarkedOpitonalAndRequired must not be marked as both optional and required"

	// +optional
	// +kubebuilder:validation:Optional
	MarkedOptionalAndKubeBuilderOptional string // want "field MarkedOptionalAndKubeBuilderOptional should use only the marker optional, kubebuilder:validation:Optional is not required"

	// +required
	// +kubebuilder:validation:Required
	MarkedRequiredAndKubeBuilderRequired string // want "field MarkedRequiredAndKubeBuilderRequired should use only the marker required, kubebuilder:validation:Required is not required"

	// +kubebuilder:validation:Optional
	KubebuilderOptionalMarker string // want "field KubebuilderOptionalMarker should use marker optional instead of kubebuilder:validation:Optional"

	// +kubebuilder:validation:Required
	KubebuilderRequiredMarker string // want "field KubebuilderRequiredMarker should use marker required instead of kubebuilder:validation:Required"

	// +kubebuilder:validation:Optional
	// +kubebuilder:validation:Optional
	MarkedWithKubeBuilderOptionalTwice string // want "field MarkedWithKubeBuilderOptionalTwice should use marker optional instead of kubebuilder:validation:Optional"

	// +kubebuilder:validation:Optional
	// +kubebuilder:validation:Optional
	// +optional
	MarkedWithKubeBuilderOptionalTwiceAndOptional string // want "field MarkedWithKubeBuilderOptionalTwiceAndOptional should use only the marker optional, kubebuilder:validation:Optional is not required"

	// +optional
	OptionalMarker string

	// +required
	RequiredMarker string
}
