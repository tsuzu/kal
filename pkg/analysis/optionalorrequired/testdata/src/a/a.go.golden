package a

type OptionalOrRequiredTestStruct struct {
	NoMarkers string // want "field NoMarkers must be marked as optional or required"

	// noOptionalOrRequiredMarker is a field with no optional or required marker.
	// +enum
	// +kubebuilder:validation:Enum=Foo;Bar
	NoOptionalOrRequiredMarker string // want "field NoOptionalOrRequiredMarker must be marked as optional or required"

	// +optional
	// +required
	MarkedOpitonalAndRequired string // want "field MarkedOpitonalAndRequired must not be marked as both optional and required"

	// +optional
	MarkedOptionalAndKubeBuilderOptional string // want "field MarkedOptionalAndKubeBuilderOptional should use only the marker optional, kubebuilder:validation:Optional is not required"

	// +required
	MarkedRequiredAndKubeBuilderRequired string // want "field MarkedRequiredAndKubeBuilderRequired should use only the marker required, kubebuilder:validation:Required is not required"

	// +optional
	KubebuilderOptionalMarker string // want "field KubebuilderOptionalMarker should use marker optional instead of kubebuilder:validation:Optional"

	// +required
	KubebuilderRequiredMarker string // want "field KubebuilderRequiredMarker should use marker required instead of kubebuilder:validation:Required"

	// +optional
	MarkedWithKubeBuilderOptionalTwice string // want "field MarkedWithKubeBuilderOptionalTwice should use marker optional instead of kubebuilder:validation:Optional"

	// +optional
	MarkedWithKubeBuilderOptionalTwiceAndOptional string // want "field MarkedWithKubeBuilderOptionalTwiceAndOptional should use only the marker optional, kubebuilder:validation:Optional is not required"

	// +optional
	OptionalMarker string

	// +required
	RequiredMarker string
}
